venom_core/services/runtime_switch_gate.py

```
@@ -0,0 +1,233 @@
+        snapshot.switch_id,
+        snapshot.source,
+        snapshot.from_runtime,
+        snapshot.to_runtime,
+    )
+    return snapshot
+
+
+def finish_runtime_switch(*, switch_id: str | None) -> None:
+    with _STATE_LOCK:
+        if switch_id and _STATE.switch_id and _STATE.switch_id != switch_id:
+            logger.warning(
```

Ruff (0.15.13)

[error] Too many arguments for `logging` format string

(PLE1205)